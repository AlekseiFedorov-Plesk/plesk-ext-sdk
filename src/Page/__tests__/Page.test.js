// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { createElement } from '@plesk/ui-library';
import Page from '../Page';
import { shallow } from 'enzyme';

describe('Page', () => {
    it('renders correctly', () => {
        const routes = [{
            path: '/',
            componentPath: 'Index',
        }];
        const wrapper = shallow(
            <Page
                path={routes[0].path}
                routes={routes}
            >
                {'Page content'}
            </Page>,
            {
                context: {
                    locale: {
                        lmsg: jest.fn(() => 'Page title'),
                    },
                },
            }
        );

        expect(wrapper).toMatchSnapshot();
    });
});
