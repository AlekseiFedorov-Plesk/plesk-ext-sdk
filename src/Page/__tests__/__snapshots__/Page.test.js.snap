// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    path="/"
    routes={
      Array [
        Object {
          "componentPath": "Index",
          "path": "/",
        },
      ]
    }
  >
    Page content
  </Page>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        "Page content",
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      "Page content",
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          "Page content",
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        "Page content",
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "locale": Object {
        "lmsg": [MockFunction] {
          "calls": Array [
            Array [
              "Index.title",
              Object {},
              null,
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": "Page title",
            },
          ],
        },
      },
    },
    "hydrateIn": undefined,
  },
}
`;
